app/uk/gov/hmrc/exports/services/mapping/governmentagencygoodsitem/AdditionalInformationBuilder.scala

```diff
@@ -43,14 +43,29 @@ class AdditionalInformationBuilder @Inject()() extends ModifyingBuilder[ExportIt
       }
     }
 
-  def buildThenAdd(exportsDeclaration: ExportsDeclaration, wcoGovernmentAgencyGoodsItem: GoodsShipment.GovernmentAgencyGoodsItem): Unit =
-    exportsDeclaration.parties.declarantIsExporter match {
-      case Some(declarantIsExporter) if declarantIsExporter.isExporter =>
-        val additionalInformation = new AdditionalInformation(code = "00400", description = "EXPORTER")
-        wcoGovernmentAgencyGoodsItem.getAdditionalInformation.add(buildAdditionalInformation(additionalInformation))
+  def buildThenAdd(
+    exportItem: ExportItem,
+    declarantIsExporter: Option[DeclarantIsExporter],
+    wcoGovernmentAgencyGoodsItem: GoodsShipment.GovernmentAgencyGoodsItem
+  ): Unit = {
+
+    val additionalInformation = AdditionalInformation(code = "00400", description = "EXPORTER")
 
+    lazy val infoEnteredByUser = exportItem.additionalInformation.exists { additionalInformationData =>
```

Comment:
thb this block is a bit convoluted.

Reply:
Agreed, could be simplified to something like:
```
lazy val infoEnteredByUser = exportItem.additionalInformation
      .map{ _.items.exists( _ == `additionalInformation`) }
      .getOrElse(false)
```

Reply:
Any reason it is defined as lazy?